### PR TITLE
A patch that implements embedding single-line contracts in single-line comments

### PR DESCRIPTION
**Pull request by x1024**
This patch makes writing backward-compatible CoffeeScript contracts possible.

### CODE REVIEW

**Comment by disnet** [comment]
Awesome!

Ran into a problem though. If you compile the webtests (via "cake build:webtests") we get a parsing error. It's because there was some commented out code (https://github.com/disnet/contracts.coffee/blob/master/test/webtest/test.coffee#L376) where the comment is getting stripped from just the contract.

Even if we fix this problem this raises a general issue with unadorned contract comment stripping. It's confusing if we are working with normal contracts and then try to comment out just the contract since it won't actually be commented out!

I think this means we should go with the `#? id :: ...` syntax.

File: test/formatting.coffee [hunk 1]
```diff
@@ -70,7 +70,7 @@ test "`?.` and `::` should continue lines", ->
   ?.foo
   #eq Object::toString, Date?.
   #prototype
-  #::
+  # ::
```
Comment (disnet) [inline test/formatting.coffee]: don't think you need this change anymore

**Comment by x1024** [comment]
Hmm.. Agreed on the commented out contracts, I hadn't thought of that. Let's go with #?
I'll update the code/tests tonight.

I had only ran "cake test".
Suggestion: let's insert webtests into "cake test"?

File: test/formatting.coffee (hunk 1 above)
Comment (x1024) [inline test/formatting.coffee]: Agreed.

**Comment by disnet** [comment]
Shoot. Just realized we have a deeper problem.

Though I said that if we disable contracts we get pure CoffeeScript semantics this is slightly not true. We have to wrap contracted values in an object with the `.use()` method. This means that if we have contracts in comment annotations code would still break in non-contracts compilers since the code would still be expecting to call `.use()`.

**Comment by disnet** [comment]
> I had only ran "cake test".
> Suggestion: let's insert webtests into "cake test"?

Done. This just compiles though. You need to load https://github.com/disnet/contracts.coffee/blob/master/test/webtest/test.html in your browser to actually run the tests.

**Comment by disnet** [comment]
Closing since the .use issue actually doesn't make it backwards compatible.

**Pull request closed without merging**